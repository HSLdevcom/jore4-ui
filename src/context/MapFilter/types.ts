import { DateTime } from 'luxon';

export enum FilterType {
  ShowFutureStops = 'show-future-stops',
  ShowCurrentStops = 'show-current-stops',
  ShowPastStops = 'show-past-stops',
}

export interface Filter {
  type: FilterType;
  enabled: boolean;
}

export interface IMapFilterContext {
  showStopFilterOverlay: boolean;
  stopFilters: Filter[];
  observationDate: DateTime;
}

export type SetStateAction = {
  type: 'setState';
  payload?: Partial<IMapFilterContext>;
};

export type SetObservationDateAction = {
  type: 'setObservationDate';
  payload: DateTime;
};

export type MapFilterActions = SetStateAction | SetObservationDateAction;
