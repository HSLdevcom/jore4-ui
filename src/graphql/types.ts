import { ApolloQueryResult, QueryResult } from '@apollo/client';

export type GqlQueryResultData = { __typename?: 'query_root' };
export type GqlMutationResultData = { __typename?: 'mutation_root' };

export type GqlQueryResult<TData extends GqlQueryResultData> =
  | QueryResult<TData> // from useQuery
  | ApolloQueryResult<TData>; // from apolloClient.query
