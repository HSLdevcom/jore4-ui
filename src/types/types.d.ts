// Trick for preventing linter warnings in places where any is needed as type
type ExplicitAny = any; // eslint-disable-line @typescript-eslint/no-explicit-any

type UUID = string;

type LocalizedString = {
  // eslint-disable-next-line camelcase
  fi_FI?: string;
  // eslint-disable-next-line camelcase
  sv_FI?: string;
};
