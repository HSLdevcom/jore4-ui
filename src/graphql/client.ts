import {
  ApolloClient,
  concat,
  HttpLink,
  InMemoryCache,
  split,
} from '@apollo/client';
import { WebSocketLink } from '@apollo/client/link/ws';
import { getMainDefinition } from '@apollo/client/utilities';
import { mapHttpToWs } from '../utils';
import { authRoleMiddleware } from './auth';

export const createGraphqlClient = () => {
  const graphqlServerUrl = '/api/graphql/v1/graphql';

  const httpLink = new HttpLink({
    uri: graphqlServerUrl,
  });

  const apolloLink = concat(authRoleMiddleware, httpLink);

  // because next.js might run this on server-side and websockets aren't
  // supported there, we have to check if we are on browser before
  // initializing WebSocket link
  const wsLink = process.browser
    ? new WebSocketLink({
        // WebSocketLink doesn't work with relative url's, so we have to
        // turn relative url into absolute.
        uri: `${mapHttpToWs(window.location.origin)}${graphqlServerUrl}`,
        options: {
          reconnect: true,
        },
      })
    : undefined;

  const link = process.browser
    ? split(
        ({ query }) => {
          const definition = getMainDefinition(query);
          const isSubscription =
            definition.kind === 'OperationDefinition' &&
            definition.operation === 'subscription';
          return isSubscription;
        },
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        wsLink!,
        apolloLink,
      )
    : apolloLink;

  const cache = new InMemoryCache({
    typePolicies: {
      route_route: {
        keyFields: ['route_id'],
      },
      service_pattern_scheduled_stop_point: {
        keyFields: ['scheduled_stop_point_id'],
      },
    },
  });

  const client = new ApolloClient({
    link,
    cache,
    defaultOptions: {
      watchQuery: {
        // Should return cached query results first while rerunning the query against the backend.
        // After the backend response has returned, the query results are updated
        fetchPolicy: 'cache-and-network',
      },
    },
  });

  return client;
};
