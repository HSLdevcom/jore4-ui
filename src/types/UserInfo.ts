export type UserInfo = {
  id: string;
  fullName?: string;
  givenName?: string;
  familyName?: string;
  permissions: string[];
};
