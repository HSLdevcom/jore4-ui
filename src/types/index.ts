export interface Point {
  latitude: number;
  longitude: number;
}

export enum Direction {
  Forward = 'forward',
  Backward = 'backward',
  BiDirectional = 'bidirectional',
}
