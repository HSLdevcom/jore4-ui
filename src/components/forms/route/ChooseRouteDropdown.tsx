import { DateTime } from 'luxon';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import {
  RouteRoute,
  useGetRouteDetailsByLabelWildcardQuery,
} from '../../../generated/graphql';
import { mapRoutesDetailsResult } from '../../../graphql';
import { MAX_DATE, MIN_DATE } from '../../../time';
import { Priority } from '../../../types/Priority';
import { Combobox, ComboboxInputProps } from '../../../uiComponents';
import { mapToVariables } from '../../../utils';
import { DateRange } from '../common/DateRange';

interface Props extends ComboboxInputProps {
  testId?: string;
  date: DateTime;
  priorities: Priority[];
}

const mapToOptionContent = (item: RouteRoute) => (
  <div className="flex flex-col">
    <div>
      <span className="font-bold">{item.label}</span>
      {` | ${item.description_i18n}`}
    </div>
    <div className="text-sm">
      <DateRange
        startDate={item.validity_start || MIN_DATE}
        endDate={item.validity_end || MAX_DATE}
      />
    </div>
  </div>
);

const mapToOption = (item: RouteRoute) => ({
  key: item.route_id,
  value: item.route_id,
  render: () => mapToOptionContent(item),
});

export const ChooseRouteDropdown = ({
  testId,
  value,
  onChange,
  onBlur,
  date,
  priorities,
}: Props): JSX.Element => {
  const { t } = useTranslation();

  const [query, setQuery] = useState('');

  const routesResult = useGetRouteDetailsByLabelWildcardQuery(
    mapToVariables({
      label: `${query}%`,
      date: date.toISO(),
      priorities,
    }),
  );

  const routes = mapRoutesDetailsResult(routesResult);

  const options = routes?.map(mapToOption) || [];

  const selectedRoute = routes?.find(
    (item) => item.route_id === value,
  ) as RouteRoute;

  // If no route is selected, show "Choose route"
  const mapToButtonContent = (displayedRoute?: RouteRoute) => {
    // Headless UI Combobox component takes care of displaying query
    if (query && !displayedRoute) return null;

    return (
      <div className="w-full">
        {displayedRoute
          ? `${displayedRoute.label} (${displayedRoute.description_i18n})`
          : t('routes.chooseRoute')}
      </div>
    );
  };

  return (
    <Combobox
      id="choose-route-combobox"
      testId={testId}
      buttonContent={mapToButtonContent(selectedRoute)}
      options={options}
      value={value}
      onChange={onChange}
      onBlur={onBlur}
      onQueryChange={setQuery}
    />
  );
};
