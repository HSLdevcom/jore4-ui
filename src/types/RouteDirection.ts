import { RouteDirectionEnum } from '../generated/graphql';

export const RouteDirection = {
  Outbound: RouteDirectionEnum.Outbound,
  Inbound: RouteDirectionEnum.Inbound,
};

export type RouteDirection =
  typeof RouteDirectionEnum[keyof typeof RouteDirection];
