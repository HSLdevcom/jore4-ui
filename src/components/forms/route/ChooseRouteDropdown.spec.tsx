import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { act, fireEvent, screen } from '@testing-library/react';
import { DateTime } from 'luxon';
import {
  GetRouteDetailsByLabelWildcardDocument,
  GetRouteDetailsByLabelWildcardQuery,
  RouteDirectionEnum,
} from '../../../generated/graphql';
import { Priority } from '../../../types/Priority';
import {
  buildLocalizedString,
  buildRoute,
  render,
  sleep,
} from '../../../utils/test-utils';
import { ChooseRouteDropdown } from './ChooseRouteDropdown';

describe(`<${ChooseRouteDropdown.name} />`, () => {
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const onChange = () => {};
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const onBlur = () => {};
  const value = '';
  const testId = 'ChooseRouteDropdown1';
  const buttonTestId = `${testId}-button`;
  const inputTestId = `${testId}-input`;

  const mocks: MockedResponse<GetRouteDetailsByLabelWildcardQuery>[] = [
    {
      request: {
        query: GetRouteDetailsByLabelWildcardDocument,
        variables: {
          label: '%',
          date: '2022-03-23T00:00:00.000+02:00',
          priorities: [10],
        },
      },
      result: {
        data: {
          route_route: [
            {
              ...buildRoute('1'),
              description_i18n: buildLocalizedString('Route 1'),
              label: '123',
              __typename: 'route_route',
              route_id: 'route1',
              validity_start: DateTime.fromISO('2022-03-20T22:00:00+00:00'),
              validity_end: DateTime.fromISO('2022-03-29T20:59:59.999+00:00'),
              priority: 10,
              direction: RouteDirectionEnum.Outbound,
              route_shape: null,
              starts_from_scheduled_stop_point_id: 'start',
              ends_at_scheduled_stop_point_id: 'end',
              on_line_id: 'line_id',
            },
            {
              ...buildRoute('2'),
              description_i18n: buildLocalizedString('Route 2'),
              label: '456',
              __typename: 'route_route',
              route_id: 'route2',
              validity_start: DateTime.fromISO('2021-01-01T22:00:00+00:00'),
              validity_end: DateTime.fromISO('2022-12-31T21:59:59.999+00:00'),
              priority: 10,
              direction: RouteDirectionEnum.Outbound,
              route_shape: null,
              starts_from_scheduled_stop_point_id: 'start',
              ends_at_scheduled_stop_point_id: 'end',
              on_line_id: 'line_id',
            },
          ],
        },
      },
    },
    {
      request: {
        query: GetRouteDetailsByLabelWildcardDocument,
        variables: {
          label: '1%',
          date: '2022-03-23T00:00:00.000+02:00',
          priorities: [10],
        },
      },
      result: {
        data: {
          route_route: [
            {
              ...buildRoute('1'),
              description_i18n: buildLocalizedString('Route 1'),
              label: '123',
              __typename: 'route_route',
              route_id: 'route1',
              validity_start: DateTime.fromISO('2022-03-20T22:00:00+00:00'),
              validity_end: DateTime.fromISO('2022-03-29T20:59:59.999+00:00'),
              priority: 10,
              direction: RouteDirectionEnum.Outbound,
              route_shape: null,
              starts_from_scheduled_stop_point_id: 'start',
              ends_at_scheduled_stop_point_id: 'end',
              on_line_id: 'line_id',
            },
          ],
        },
      },
    },
  ];

  test('Opens dropdown when clicked and shows all routes', async () => {
    const { container, asFragment } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ChooseRouteDropdown
          testId={testId}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          date={DateTime.fromISO('2022-03-23T00:00:00.000+02:00')}
          priorities={[Priority.Standard]}
        />
      </MockedProvider>,
    );

    // dropdown is collapsed, value is not yet bound to dropdown, 'Choose route' text shows
    expect(screen.getByTestId(buttonTestId)).toHaveTextContent(
      'Valitse reitti',
    );
    expect(container.querySelector('li')).toBeNull();
    expect(asFragment()).toMatchSnapshot();

    // wait for the graphql call to execute
    await act(() => sleep(0));

    // dropdown is collapsed, data is loaded and 'Choose route' text still shows in button
    expect(screen.getByTestId(buttonTestId)).toHaveTextContent(
      'Valitse reitti',
    );
    expect(container.querySelector('li')).toBeNull();
    expect(asFragment()).toMatchSnapshot();

    // click dropdown to open it:
    const openDropdownButton = screen.getByTestId(buttonTestId);
    fireEvent.click(openDropdownButton);

    // dropdown is open, both lines show
    const items = container.querySelectorAll('li');

    expect(items[0].querySelectorAll('div > div')[0].textContent).toBe(
      `123 | route 1`,
    );
    expect(items[0].querySelectorAll('div > div')[1].textContent).toBe(
      `21.3.2022 - 29.3.2022`,
    );
    expect(items[1].querySelectorAll('div > div')[0].textContent).toBe(
      `456 | route 2`,
    );
    expect(items[1].querySelectorAll('div > div')[1].textContent).toBe(
      `2.1.2021 - 31.12.2022`,
    );
    expect(items.length).toBe(2);

    expect(asFragment()).toMatchSnapshot();
  });

  test('Filters shown routes when query is set', async () => {
    const { container, asFragment } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ChooseRouteDropdown
          testId={testId}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          date={DateTime.fromISO('2022-03-23T00:00:00.000+02:00')}
          priorities={[Priority.Standard]}
        />
      </MockedProvider>,
    );

    // dropdown is collapsed, value is not yet bound to dropdown, 'Choose route' text shows
    expect(screen.getByTestId(buttonTestId)).toHaveTextContent(
      'Valitse reitti',
    );
    expect(container.querySelector('li')).toBeNull();
    expect(asFragment()).toMatchSnapshot();

    // wait for the graphql call to execute
    await act(() => sleep(0));

    // dropdown is collapsed, data is loaded and 'Choose route' text still shows in button
    expect(screen.getByTestId(buttonTestId)).toHaveTextContent(
      'Valitse reitti',
    );
    expect(container.querySelector('li')).toBeNull();
    expect(asFragment()).toMatchSnapshot();

    // write query
    const input = screen.getByTestId(inputTestId);
    fireEvent.change(input, { target: { value: '1' } });

    // wait for the graphql call to execute
    await act(() => sleep(0));

    // dropdown is open, first line shows
    const items = container.querySelectorAll('li');

    expect(items[0].querySelectorAll('div > div')[0].textContent).toBe(
      `123 | route 1`,
    );
    expect(items[0].querySelectorAll('div > div')[1].textContent).toBe(
      `21.3.2022 - 29.3.2022`,
    );
    expect(items.length).toBe(1);

    expect(asFragment()).toMatchSnapshot();
  });
});
