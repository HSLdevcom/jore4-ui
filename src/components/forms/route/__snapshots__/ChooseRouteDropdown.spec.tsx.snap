// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseRouteDropdown /> Filters shown routes when query is set 1`] = `
<DocumentFragment>
  <div
    class="relative w-full"
    data-testid="ChooseRouteDropdown1"
    id="choose-route-combobox"
  >
    <input
      aria-expanded="false"
      class="input-element relative h-full w-full"
      data-testid="ChooseRouteDropdown1-input"
      id="headlessui-combobox-input-7"
      role="combobox"
      type="text"
    />
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="absolute inset-y-0 right-0 flex h-full w-full items-center justify-end px-3 text-left focus:outline-none"
      data-testid="ChooseRouteDropdown1-button"
      id="headlessui-combobox-button-8"
      tabindex="-1"
      type="button"
    >
      <div
        class="w-full"
      >
        Valitse reitti
      </div>
      <svg
        class="mr-1 ml-auto text-2xl text-tweaked-brand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <i
        class="icon-arrow text-tweaked-brand transition duration-150 ease-in-out rotate-0"
        style="font-size: 10px;"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`<ChooseRouteDropdown /> Filters shown routes when query is set 2`] = `
<DocumentFragment>
  <div
    class="relative w-full"
    data-testid="ChooseRouteDropdown1"
    id="choose-route-combobox"
  >
    <input
      aria-expanded="false"
      class="input-element relative h-full w-full"
      data-testid="ChooseRouteDropdown1-input"
      id="headlessui-combobox-input-7"
      role="combobox"
      type="text"
    />
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="absolute inset-y-0 right-0 flex h-full w-full items-center justify-end px-3 text-left focus:outline-none"
      data-testid="ChooseRouteDropdown1-button"
      id="headlessui-combobox-button-8"
      tabindex="-1"
      type="button"
    >
      <div
        class="w-full"
      >
        Valitse reitti
      </div>
      <svg
        class="mr-1 ml-auto text-2xl text-tweaked-brand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <i
        class="icon-arrow text-tweaked-brand transition duration-150 ease-in-out rotate-0"
        style="font-size: 10px;"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`<ChooseRouteDropdown /> Filters shown routes when query is set 3`] = `
<DocumentFragment>
  <div
    class="relative w-full"
    data-testid="ChooseRouteDropdown1"
    id="choose-route-combobox"
  >
    <input
      aria-controls="headlessui-combobox-options-10"
      aria-expanded="true"
      class="input-element relative h-full w-full"
      data-testid="ChooseRouteDropdown1-input"
      id="headlessui-combobox-input-7"
      role="combobox"
      type="text"
    />
    <button
      aria-controls="headlessui-combobox-options-10"
      aria-expanded="true"
      aria-haspopup="true"
      class="absolute inset-y-0 right-0 flex h-full w-full items-center justify-end px-3 text-left focus:outline-none"
      data-testid="ChooseRouteDropdown1-button"
      id="headlessui-combobox-button-8"
      tabindex="-1"
      type="button"
    >
      <svg
        class="mr-1 ml-auto text-2xl text-tweaked-brand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <i
        class="icon-arrow text-tweaked-brand transition duration-150 ease-in-out -rotate-180"
        style="font-size: 10px;"
      />
    </button>
    <ul
      aria-labelledby="headlessui-combobox-button-8"
      class="absolute left-0 z-10 w-full rounded-b-md border border-black border-opacity-20 bg-white shadow-md focus:outline-none transition ease-out duration-100 opacity-0 scale-95"
      id="headlessui-combobox-options-10"
      role="listbox"
    >
      <li
        id="headlessui-combobox-option-11"
        role="option"
        tabindex="-1"
      >
        <span
          class=" flex group text-left pr-2 py-2 focus:outline-none items-center"
        >
          <svg
            class="mx-1 text-2xl invisible"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
          <div
            class="flex flex-col"
          >
            <div>
              <span
                class="font-bold"
              >
                123
              </span>
               | Route 1
            </div>
            <div
              class="text-sm"
            >
              21.3.2022 - 29.3.2022
            </div>
          </div>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ChooseRouteDropdown /> Opens dropdown when clicked and shows all routes 1`] = `
<DocumentFragment>
  <div
    class="relative w-full"
    data-testid="ChooseRouteDropdown1"
    id="choose-route-combobox"
  >
    <input
      aria-expanded="false"
      class="input-element relative h-full w-full"
      data-testid="ChooseRouteDropdown1-input"
      id="headlessui-combobox-input-1"
      role="combobox"
      type="text"
    />
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="absolute inset-y-0 right-0 flex h-full w-full items-center justify-end px-3 text-left focus:outline-none"
      data-testid="ChooseRouteDropdown1-button"
      id="headlessui-combobox-button-2"
      tabindex="-1"
      type="button"
    >
      <div
        class="w-full"
      >
        Valitse reitti
      </div>
      <svg
        class="mr-1 ml-auto text-2xl text-tweaked-brand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <i
        class="icon-arrow text-tweaked-brand transition duration-150 ease-in-out rotate-0"
        style="font-size: 10px;"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`<ChooseRouteDropdown /> Opens dropdown when clicked and shows all routes 2`] = `
<DocumentFragment>
  <div
    class="relative w-full"
    data-testid="ChooseRouteDropdown1"
    id="choose-route-combobox"
  >
    <input
      aria-expanded="false"
      class="input-element relative h-full w-full"
      data-testid="ChooseRouteDropdown1-input"
      id="headlessui-combobox-input-1"
      role="combobox"
      type="text"
    />
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="absolute inset-y-0 right-0 flex h-full w-full items-center justify-end px-3 text-left focus:outline-none"
      data-testid="ChooseRouteDropdown1-button"
      id="headlessui-combobox-button-2"
      tabindex="-1"
      type="button"
    >
      <div
        class="w-full"
      >
        Valitse reitti
      </div>
      <svg
        class="mr-1 ml-auto text-2xl text-tweaked-brand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <i
        class="icon-arrow text-tweaked-brand transition duration-150 ease-in-out rotate-0"
        style="font-size: 10px;"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`<ChooseRouteDropdown /> Opens dropdown when clicked and shows all routes 3`] = `
<DocumentFragment>
  <div
    class="relative w-full"
    data-testid="ChooseRouteDropdown1"
    id="choose-route-combobox"
  >
    <input
      aria-controls="headlessui-combobox-options-4"
      aria-expanded="true"
      class="input-element relative h-full w-full"
      data-testid="ChooseRouteDropdown1-input"
      id="headlessui-combobox-input-1"
      role="combobox"
      type="text"
    />
    <button
      aria-controls="headlessui-combobox-options-4"
      aria-expanded="true"
      aria-haspopup="true"
      class="absolute inset-y-0 right-0 flex h-full w-full items-center justify-end px-3 text-left focus:outline-none"
      data-testid="ChooseRouteDropdown1-button"
      id="headlessui-combobox-button-2"
      tabindex="-1"
      type="button"
    >
      <div
        class="w-full"
      >
        Valitse reitti
      </div>
      <svg
        class="mr-1 ml-auto text-2xl text-tweaked-brand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <i
        class="icon-arrow text-tweaked-brand transition duration-150 ease-in-out -rotate-180"
        style="font-size: 10px;"
      />
    </button>
    <ul
      aria-labelledby="headlessui-combobox-button-2"
      class="absolute left-0 z-10 w-full rounded-b-md border border-black border-opacity-20 bg-white shadow-md focus:outline-none transition ease-out duration-100 opacity-0 scale-95"
      id="headlessui-combobox-options-4"
      role="listbox"
    >
      <li
        id="headlessui-combobox-option-5"
        role="option"
        tabindex="-1"
      >
        <span
          class=" flex group text-left pr-2 py-2 focus:outline-none items-center"
        >
          <svg
            class="mx-1 text-2xl invisible"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
          <div
            class="flex flex-col"
          >
            <div>
              <span
                class="font-bold"
              >
                123
              </span>
               | Route 1
            </div>
            <div
              class="text-sm"
            >
              21.3.2022 - 29.3.2022
            </div>
          </div>
        </span>
      </li>
      <li
        id="headlessui-combobox-option-6"
        role="option"
        tabindex="-1"
      >
        <span
          class=" flex group text-left pr-2 py-2 focus:outline-none items-center"
        >
          <svg
            class="mx-1 text-2xl invisible"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
          <div
            class="flex flex-col"
          >
            <div>
              <span
                class="font-bold"
              >
                456
              </span>
               | Route 2
            </div>
            <div
              class="text-sm"
            >
              2.1.2021 - 31.12.2022
            </div>
          </div>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
